neuraxle/metaopt/random.py

```diff
@@ -377,3 +383,166 @@ def transform(self, data_inputs):
         if self.best_validation_wrapper_of_model is None:
             raise Exception('Cannot transform RandomSearch before fit')
         return self.best_validation_wrapper_of_model.wrapped.transform(data_inputs)
+
+
+class HyperparamsRepository(ABC):
+    """
+    Hyperparams repository that saves hyperparams, and scores for every AutoML trial.
+
+    .. seealso::
+        :class:`HyperparamsRepository`,
+        :class:`HyperparameterSamples`,
+        :class:`AutoMLSequentialWrapper`
+    """
+
+    @abstractmethod
+    def create_new_trial(self, hyperparams: HyperparameterSamples):
+        """
+        Create new hyperperams trial.
+
+        :param hyperparams: hyperparams
+        :type hyperparams: HyperparameterSamples
+        :return:
+        """
+        pass
+
+    @abstractmethod
+    def load_all_trials(self) -> Tuple[List[HyperparameterSamples], List[float]]:
+        """
+        Load all hyperparameter trials with their corresponding score.
+
+        :return: (hyperparams, scores)
+        """
+        pass
+
+    @abstractmethod
+    def save_score_for_trial(self, hyperparams: HyperparameterSamples, score: float):
+        """
+        Save hyperparams, and score for a trial.
+
+        :return: (hyperparams, scores)
+        """
+        pass
+
+    def _get_trial_hash(self, hp_dict):
+        current_hyperparameters_hash = hashlib.md5(str.encode(str(hp_dict))).hexdigest()
+        return current_hyperparameters_hash
+
+
+class HyperparamsJSONRepository(HyperparamsRepository):
+    """
+    Hyperparams repository that saves json files for every AutoML trial.
+
+    .. seealso::
+        :class:`HyperparamsRepository`,
+        :class:`HyperparameterSamples`,
+        :class:`AutoMLSequentialWrapper`
+    """
+
+    def __init__(self, folder):
+        self.cache_folder = folder
```

Comment:
Should we instead pass a context to `HyperparamsRepository.__init__`?

Can we train many things in the same context without problems?

We could use a subfolder of the context aside the pipeline.